packages/compile-vyper/index.js

```diff
@@ -119,7 +119,7 @@ const Compile = {
     const vyperFiles = sources.filter(path => minimatch(path, VYPER_PATTERN));
 
     // no vyper files found, no need to check vyper
-    if (sources.length === 0) {
+    if (vyperFiles.length === 0) {
```

Comment:
Unrelated to the work in this PR, turns out this check was just wrong. (Also should be own PR)

Reply:
(Opened #3384 for this change)